modules/services/window-managers/hyprland.nix

```diff
@@ -175,22 +175,34 @@ in {
       shouldGenerate = cfg.systemdIntegration || cfg.extraConfig != ""
         || combinedSettings != { };
 
-      toHyprconf = attrs:
+      toHyprconf = with lib;
+        attrs: indentLevel:
         let
+          indent = concatStrings (replicate indentLevel "  ");
+
           mkSection = n: attrs: ''
-            ${n} {
-              ${toHyprconf attrs}}
+            ${indent}${n} {
+            ${toHyprconf attrs (indentLevel + 1)}${indent}}
           '';
-          mkFields = lib.generators.toKeyValue { listsAsDuplicateKeys = true; };
-          sections = lib.filterAttrs (n: v: lib.isAttrs v) attrs;
-          fields = lib.filterAttrs (n: v: !(lib.isAttrs v)) attrs;
-        in lib.concatStringsSep "\n" (lib.mapAttrsToList mkSection sections)
-        + mkFields fields;
+          sections = filterAttrs (n: v: isAttrs v) attrs;
+
+          mkFields = generators.toKeyValue { listsAsDuplicateKeys = true; };
+          allFields = filterAttrs (n: v: !(isAttrs v)) attrs;
+          importantFields =
+            filterAttrs (n: _: (hasPrefix "$" n) || (hasPrefix "bezier" n))
```

Comment:
A bit late, but shouldn't [`source`](https://wiki.hyprland.org/Configuring/Keywords/#sourcing-multi-file) also be included here?

Reply:
I think in some cases it might break users' configs.